osu.Server.Spectator/Database/DatabaseAccess.cs

```diff
@@ -335,6 +336,24 @@ public async Task<bool> GetUserAllowsPMs(int userId)
             });
         }
 
+        public async Task<IEnumerable<osu_build>> GetAllLazerBuildsAsync()
+        {
+            var connection = await getConnectionAsync();
+
+            return await connection.QueryAsync<osu_build>(
+                "SELECT `build_id`, `version`, `hash`, `users` "
+                + "FROM `osu_builds` `b` "
+                + "JOIN `osu_updates`.`streams` `s` ON `b`.`stream_id` = `s`.`stream_id` "
+                + "WHERE `s`.`name` = 'lazer'");
+        }
```

Comment:
If there's anything that needs scrutinising in this diff, it is this query. I am  _reasonably_ convinced that it will work in production, but it would be wise to check, especially that we have [weird carve-outs for android still](https://github.com/ppy/osu/blob/07da9d95a90c330b4bf90e26fb68ee24a2e3e803/osu.Game/OsuGameBase.cs#L256-L266) and I'm not sure how that works in production.

Generally the reason this is a thing is to decrease blast radius, first of all, as I generally wouldn't want to be spraying updates on production based from a client-supplied version hash without any checking. However it has a correctness purpose as well - see the implementation of `BuildUserCountUpdater`. If the thing was to only rely on the user counts as reported by clients to the server, it would stop being aware of a build once the last client that was connected to it on that build disconnects.

Reply:
Modified slightly, seems fine otherwise.

Obviously won't hold out if we horizontally scale, but we'll deal with that later.

Reply:
Yeah, I have done 0 consideration for scaling here given that the rest of the server doesn't scale.